src/background/sidebar-injector.js

```diff
@@ -80,18 +108,18 @@ export default function SidebarInjector(
    *
    * @param {chrome.tabs.Tab} tab - A tab object representing the tab to insert the sidebar
    *        into.
-   * @param {Object?} config - An object containing configuration info that
+   * @param {object} config - An object containing configuration info that
    *        is passed to the app when it loads.
    *
    * Returns a promise that will be resolved if the injection succeeded
    * otherwise it will be rejected with an error.
    */
-  this.injectIntoTab = function (tab, config) {
-    config = config || {};
-    if (isFileURL(tab.url)) {
-      return injectIntoLocalDocument(tab);
+  this.injectIntoTab = function (tab, config = {}) {
+    const tab_ = checkTab(tab);
```

Comment:
> Are you 100% confident that you want to throw an exception here if the tab does not have an id and url?

Yes. If this function throws an error it will cause the browser action to display an "!" error indicator in its badge and the user can click it to see why the client failed to inject. You can see this in action if you open a tab at chrome://settings for example and try to activate the client.